### ISSUE TITLE
Snowflake: dynamic table column definition with WITH TAG but no data type is reported as unparsable

### ISSUE THREAD

**Issue by inoue-koki** [sqlfluff/sqlfluff#7753]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When linting or parsing Snowflake DDL for a dynamic table, SQLFluff reports Found unparsable section if a column definition omits the data type and only uses WITH TAG.

This seems to affect column definitions in CREATE DYNAMIC TABLE specifically. Similar syntax works in other cases, such as:

* when the column includes an explicit data type before WITH TAG
* when the column only has COMMENT
* when the same WITH TAG syntax is used in CREATE VIEW

Because Snowflake allows these column definitions, I believe this is a parser bug or missing syntax support in the Snowflake dialect.

### Expected Behaviour

SQLFluff should parse this Snowflake statement without reporting an unparsable section.

In particular, a column definition in CREATE DYNAMIC TABLE should be accepted when it omits the data type and only includes WITH TAG.

### Observed Behaviour

SQLFluff reports:

Found unparsable section

This happens for the following SQL:

```sql
create dynamic table test (
    col with tag (tag_name = 'tag_value')
)
warehouse = dev_wh
target_lag = 'DOWNSTREAM'
as
select
    col
from base_table
```

The following statements do not raise the same error:

```sql
create dynamic table test (
    col varchar(10) with tag (tag_name = 'tag_value')
)
warehouse = dev_wh
target_lag = 'DOWNSTREAM'
as
select
    col
from base_table
```

```sql
create dynamic table test (
    col comment 'wwwww'
)
warehouse = dev_wh
target_lag = 'DOWNSTREAM'
as
select
    col
from base_table
```

```sql
create view test (
    col with tag (tag_value = 'tag_name')
)
as
select
    col
from base_table
```

### How to reproduce

1. Create a file such as repro.sql with the following contents:

```sql
create dynamic table test (
    col with tag (tag_name = 'tag_value')
)
warehouse = dev_pipeline_wh
target_lag = 'DOWNSTREAM'
as
select
    col
from base_table
```

2. Run SQLFluff with the Snowflake dialect, for example:

```bash
sqlfluff parse --dialect snowflake repro.sql
```

3. Observe that SQLFluff reports Found unparsable section.

### Dialect

snowflake

### Version

sqlfluff: 4.1.0
python: 3.13.12

### Configuration

[sqlfluff]
dialect = snowflake
max_line_length = 150
exclude_rules = AM02,CV02,LT09,RF04,ST01,ST02,ST06

[sqlfluff:indentation]
indent_unit = space
indented_joins = True
indented_on_contents = False
indented_then = False

[sqlfluff:layout:type:select_clause]
keyword_line_position = alone
[sqlfluff:layout:type:where_clause]
keyword_line_position = alone

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
unquoted_identifiers_policy = table_aliases

[sqlfluff:rules:aliasing.table]
aliasing = implicit

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
